packages/query-core/src/removable.ts

```diff
@@ -11,7 +11,7 @@ export abstract class Removable {
   protected scheduleGc(): void {
     this.clearGcTimeout()
 
-    if (isValidTimeout(this.gcTime)) {
+    if (isValidTimeout(this.gcTime) || this.gcTime === 0) {
```

Comment:
note: this was the _only_ place where we wanted 0 to be a valid timeout, so I re-added the extra check here. Otherwise, gcTime 0 would not garbage collect.